Phase_01/Phase_01_Full_Text_Search/src/InvertedIndex.java

```diff
@@ -4,41 +4,55 @@
 import java.util.Set;
 
 public class InvertedIndex {
-    private Hashtable<String, Set<Integer>> inverted_map;
-    private ArrayList<Set<String>> books;
 
-    public  InvertedIndex(ArrayList<Set<String>> books){
-        this.books = books;
-        this.inverted_map = new Hashtable<String, Set<Integer>>();
-        CalculateIndexes();
+    /**
+     * The inverted index hashtable.
+     */
+    private final Hashtable<String, Set<Integer>> indexMap;
```

Comment:
based on https://stackoverflow.com/questions/40471/what-are-the-differences-between-a-hashmap-and-a-hashtable-in-java it's better to use hashmap instead of hashtable